README.md

```diff
@@ -1,75 +1,89 @@
 
 ### Waterfall function:
+
 The waterfall function takes three arguments:
 
-1) An initial argument, this is passed into the first async function in the waterfall.
+1. An initial argument, this is passed into the first async function in the waterfall.
 
-2) The second argument is an array of functions, each function takes two arguments. The first argument it uses to find a result and the second is a callback function.
+2. The second argument is an array of functions, each function takes two arguments. The first argument it uses to find a result and the second is a callback function.
 
-3) A final callback is given as the third argument. This is called after the waterfall has invoked all of the functions in the array.
+3. A final callback is given as the third argument. This is called after the waterfall has invoked all of the functions in the array.
 
-### Workshop: 
+### Workshop:
 
-1) Copy the code below and paste it into [Repl](https://repl.it/languages/javascript).
+1. Copy the code below and paste it into [Repl](https://repl.it/languages/javascript).
 
-2) Create the waterfall function, pass the tests. 
+2. Create the waterfall function, pass the tests.
 
-3) For bonus points, create some additional tests for the waterfall function.
+3. For bonus points, create some additional tests for the waterfall function.
 
-```
+```js
 function asyncAddOne(x, callBack) {
   setTimeout(function() {
-    if (typeof x !== 'number'){ return callBack(new Error('need a number!')) }
-    else { return callBack(null, x + 1) }
-  }, 200)
+    return callBack(x + 1);
+  }, 200);
 }
 
 function asyncDouble(x, callBack) {
   setTimeout(function() {
-    if (typeof x !== 'number') { return callBack(new Error('need a number!')) }
-    else { return callBack(null, x * 2) }
-  }, 200)
+    return callBack(x * 2);
+  }, 200);
 }
 
 function asyncTimesTen(x, callBack) {
   setTimeout(function() {
-    if (typeof x !== 'number') { return callBack(new Error('need a number!')) }
-    else { return callBack(null, x * 10) }
-  }, 200)
+    return callBack(x * 10);
+  }, 200);
 }
 
 // Create this function!
 function waterfall(arg, tasks, cb) {
-  cb(new Error('waterfall function not implemented'));
+  cb(null);
 }
 
-waterfall(3, [
-  asyncAddOne,
-  asyncDouble,
-  asyncTimesTen
-], function(error, result) {
+waterfall(3, [asyncAddOne, asyncDouble, asyncTimesTen], function(result) {
   console.log('Test 1');
-  if (error) {
-    console.log('test failed, ' + error)
-  }
-  else if (result !== 80) {
-    console.log('test failed, expected 80 but got', result)
+  if (result !== 80) {
+    console.log('test failed, expected 80 but got', result);
+  } else {
+    console.log('Test 1 passed!');
   }
-  else {
-    console.log('Test 1 passed!')
-  }
-})
+});
 ```
 
+Waterfall will apply the previous function's result to the next, for example:
+
+`waterfall(argument, [fn1, fn2, fn3,...], cb)` will return :
+
+`cb(fn_n(fn_n-1(...fn3(fn2(fn1(argument))))))`
```

Comment:
This is the synchronous version (essentially `compose`). Is it confusing to imply that this is what `waterfall` does (only to then follow it with the real example of what `waterfall` does?)

Reply:
tbh, wasn't really sure which format would be clearer for people to understand what it does :sweat_smile: 
What do you think?

Reply:
I suspect it would be less confusing if this line were removed, but I'm happy for you to leave it in and see how it plays in Nazareth if you like?

Reply:
Yeah, let's see how it goes